backend/services/esweService.ts

```diff
@@ -120,44 +106,46 @@ export default class EsweService {
     return createFlaggedContent(content)
   }
 
-  async getLatestLearningDifficulty(nomisId: string): Promise<FeatureFlagged<string>> {
+  async getLearningDifficulties(nomisId: string): Promise<LearningDifficulties> {
     if (!app.esweEnabled) {
       return createFlaggedContent(null)
     }
 
-    let content: string = null
     try {
       const context = await this.systemOauthClient.getClientCredentialsTokens()
-
-      // TODO: speak with curious to see if filtering/sorting can be added to their api
-      const [educations, profiles] = await Promise.all([
-        this.curiousApi.getLearnerEducation(context, nomisId),
-        this.curiousApi.getLearnerProfiles(context, nomisId),
-      ])
-
-      const compareByDateDesc = (a: curious.LearnerEducation, b: curious.LearnerEducation) =>
-        compareByDate(parseDate(a.learningStartDate), parseDate(b.learningStartDate))
-
-      educations.sort(compareByDateDesc)
-
-      const profilesByEstablishment = profiles.reduce(
-        (acc, current) => ({
-          ...acc,
-          [current.establishmentId]: current,
-        }),
-        {}
-      )
-
-      const lddHealthProblem = educations
-        .map(({ establishmentId }) => profilesByEstablishment[establishmentId]?.lddHealthProblem)
-        .find((value) => !!value)
-
-      content = lddHealthProblem ?? ''
+      const profiles = await this.curiousApi.getLearnerProfiles(context, nomisId)
+
+      const LddList = []
+      if (Array.isArray(profiles) && profiles.length > 0) {
```

Comment:
type checking - can we not use typescript (and maybe a null/undefined check) to make sure? Maybe not.. but I thought we could